cterasdk/edge/directorytree.py

```diff
@@ -0,0 +1,142 @@
+        root._parent = None  # pylint: disable=protected-access
+        nodes = [root]
+        while nodes:
+            node = nodes.pop(0)
+            if self._has_children(node):
+                for child_node in node.children:
+                    child_node._parent = node  # pylint: disable=protected-access
+                    nodes.append(child_node)
+        return root
+
+    def include_folder(self, path):
+        self._configure_selection(path, True, True)
+
+    def include_file(self, path):
+        self._configure_selection(path, False, True)
+
+    def exclude_folder(self, path):
+        self._configure_selection(path, True, False)
+
+    def exclude_file(self, path):
+        self._configure_selection(path, False, False)
+
+    def _configure_selection(self, path, is_dir, include):
+        self._check_root_dir(path)
+        node, parts = self._lookup(path)
+        if not parts:
+            if is_dir and not self._is_dir(node):
+                raise CTERAException('Expected to find a directory but found file', None, path=path)
+            if self._is_dir(node) and not is_dir:
+                raise CTERAException('Expected to find a file but found a directory', None, path=path)
+            node.isIncluded = include
+            node.children = None
+        else:
+            self._populate_selection(node, parts, is_dir, include)
+
+    def remove_selection(self, path):
+        self._check_root_dir(path)
+        node, parts = self._lookup(path)
+        if not parts:
+            if node._parent is not None:  # pylint: disable=protected-access
+                self._remove_child(node._parent, node.name)  # pylint: disable=protected-access
+            else:
+                node.isIncluded = False
+                node.children = None
+        else:
+            raise CTERAException('Could not find directory path', None, path=path)
+
+    def select_all(self):
+        self.root.isIncluded = True
+        self.root.children = None
+
+    def unselect_all(self):
+        self.root.isIncluded = False
+        self.root.children = None
+
+    def _check_root_dir(self, path):
+        parts = path.split('/')
+        if self.root.name != parts[0]:
+            raise CTERAException('Invalid root directory', None, input=parts[0], should_start_with=self.root.name)
+
+    def _lookup(self, path):
+        parts = path.split('/')[1:]
+        node = self.root
+        while parts:
+            if self._has_children(node):
+                child_name = parts.pop(0)
+                child_node = self._get_child(node, child_name)
+                if child_node is None:
+                    parts.insert(0, child_name)
+                    break
+                node = child_node
+            else:
+                break
+        return (node, parts)
+
+    # pylint: disable=R0201
+    def _is_dir(self, node):
+        return not hasattr(node, '_classname') or node._classname == 'DirEntry'  # pylint: disable=protected-access
```

Comment:
You can replace it with a single check:
return getattr(node, '_classname', 'DirEntry') == 'DirEntry'
This way if the attribute does not exist, getattr will return 'DirEntry' and the check will be true
It will also remove the pylint disable

Reply:
Done